Automattic/woocommerce-subscriptions-core
# Rename and refactor the Related Order Data Store classes to remove WP Post references

Fixes #145 

## Description

Don't be alarmed by the large changeset of this PR 😅 It seems like there's a lot changing, however, there's not that much. Here's a breakdown:

1. Rename the two `Related_Order_Data_Store(_Cached)_CPT` classes 
2. Remove WP Post API calls to fetch/update WC Orders in these classes

## Rename the two `Related_Order_Data_Store(_Cached)_CPT` classes

The `WCS_Related_Order_Store_CPT` and `_Cached` classes, by name, are associated heavily with Orders being a custom post type. With WC Orders moving to custom order tables we needed to rethink how this class is named and how it interacts with orders to remove any assumptions that orders are stored in the posts table.

In this PR I've removed the "CPT" (custom post type) reference in the classname and file name of both related order data store classes, and moved copies of the old classes to `/includes/deprecated`.

What the changes include:
1. Copied the `WCS_Related_Order_Store_Cached_CPT` class/file and renamed one of them to `WCS_Related_Order_Data_Store_Cached`
2. Duplicated the `WCS_Related_Order_Store_CPT` calss/file and renamed one to `WCS_Related_Order_Data_Store`
3. Moved copies/olds classes to `includes/deprecated` to maintain backward compatibility
4. Update the autoloader to locate the new classes but also located the old `_CPT` classes in the deprecated directory
5. Updated `WCS_Related_Order_Store::instance()` to use new `WCS_Related_Order_Data_Store_Cached` class
6. Deprecate all the old functions

## Remove WP Post API calls to fetch/update WC Orders

The second half of the changes made in this PR was to remove any `get_posts()`, `get_post_meta()`, `update_post_meta()` or any other WP Post API function calls from the new `WCS_Related_Order_Data_Store/_Cached` classes.

When Orders are moved to custom tables, these functions won't work so they need to be updated.

The biggest change needed was to the [`get_related_order_ids()`](https://github.com/Automattic/woocommerce-subscriptions-core/blob/4c6d848307a37c8886aaaf64ad1ad9911245682c/includes/data-stores/class-wcs-related-order-data-store.php#L44) function. This function used to use `get_posts()` with a meta query to get a subscriptions related orders by `_subscriptions_{relation_type}` meta.







## How to test this PR





The main area of subscriptions that are affected by these changes is:

1. Viewing related order tables on the Edit Subscription page (which fetches order IDs from the `_cached` meta)
   1. Making sure the new renewal order is visible in table
   2. Deleting a renewal order updates the cache
   3. Updating/deleting the `_subscription_renewal` order meta updates the subscriptions related order cached value (doesn't work if you delete meta directly via DB)
2. Viewing a subscription from My Account > Subscriptions > View #
   1. Making sure all renewal, resubscribe, switch orders are present. 

To test this PR I went over those affected areas and made sure everything was working as expected

## Product impact
